### ISSUE TITLE
Run the cluster failure analysis for Nilearn

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4165]
Chatted with @jeromedockes about this on slack

Anyone knows if the cluster Failure procedure was applied on the nilearn codebase?

https://www.pnas.org/doi/10.1073/pnas.1602413113

Basically running GLM with dummy events / task on resting state data to check that the false positive rate are indeed in the expected range.

Was looking for some "system tests" to help validate some things in nilearn and that would be a good one.

@jeromedockes mentioned that this could also spawn some "tools" to help users evaluate their pipelines.

**Comment by jeromedockes** [comment]
I'll ping @michellewang and @mtorabi59 who might be interested :)

**Comment by bthirion** [comment]
I'm not aware of any such initiative so far. 
Note that we 've put forward the non-parametric tests, which, by construction, should be "cluster-failure-proof".

**Comment by jeromedockes** [comment]
that's true but regardless of cluster failure I think it might be nice to have some utilities for sanity check / pipeline inspection report / integration test that shows how a pipeline behaves on null data -- number of false discoveries for glm, accuracy for decoding, etc.

**Comment by Remi-Gau** [comment]
> which, by construction, should be "cluster-failure-proof"

I think I may be paranoid but the "should" is doing the heavy lifting (assuming our code is good and will stay good). If we have some ways of showing that it is indeed the case then it's better.

**Comment by bthirion** [comment]
:-)
 AFAICT permutation tests are still taken as a Gold standard, but the devil is in the details. The only twist is that when you use cluster-level inference of TFCE, there are still some assumptions, such as spatial stationarity of the maps.

**Comment by michellewang** [comment]
I can take a look/work on this in the new year!
